src/presign/participant.rs

```diff
@@ -422,21 +416,60 @@ impl PresignParticipant {
         message: &Message,
         input: &Input,
     ) -> Result<ProcessOutcome<<Self as ProtocolParticipant>::Output>> {
-        info!("Handling round one presign message.");
+        info!("Presign: Handling round one message.");
 
-        // Check if we have both have received the broadcasted ciphertexts that we need
-        // in order to respond and have started round one
-        if !(self
+        // First check that we have the round one public broadcast from this
```

Comment:
todo: following up other comment on "ready" messages, I think we also need a check that the ready round is complete.

Reply:
It actually looks like currently the "ready" messages aren't strictly necessary (and this applies to all the protocols): suppose we get r1 messages from all the parties. We then proceed to `gen_round_two_msgs`, which first checks whether we've generated our own r1 message. If not, we run `gen_round_one_msgs`. It appears this triggers regardless of what happens with the ready messages.

If I do add a check that `ready` has completed, the protocol hangs, which makes me wonder if `ready` messages haven't been really doing their job all along...

Reply:
Hm, that's interesting. I feel like ready messages are a bit of an attempt to solve a much bigger deployment problem (how do you synchronize initialization of all the parties so that no messages get lost?) that I'd rather offload onto the calling application. Anyway, we can leave it for now.